src/components/text-input/TextInput.js

```diff
@@ -49,10 +50,18 @@ const TextInput = ({
     labelTooltip,
     ...rest
 }: Props) => {
+    const hasError = !!error;
     const classes = classNames(className, 'text-input-container', {
-        'show-error': !!error,
+        'show-error': hasError,
     });
 
+    const errorMessageID = React.useRef(uniqueId('errorMessage')).current;
```

Comment:
To avoid bumping that counter too much we can do this 

```jsx
// above the comp
const createInstanceId = memoize(uniqueId)

// the comp
const X = (props) => {
  const instance = useRef()
  const id = createInstanceId(instance)
}
```

Reply:
I can set that up in a followup PR if it becomes a problem, but honestly if it comes to that, I think it's more directly understandable as a class component with an instance property.